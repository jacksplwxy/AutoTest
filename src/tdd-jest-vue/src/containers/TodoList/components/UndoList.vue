<template>
  <div>
    <div data-test="count">{{ list.length }}</div>
    <ul>
      <li v-for="(item, index) in list" :key="index" data-test="item" style="display:flex;">
        {{ item }}
        <span
          data-test="delete-button"
          @click="
            () => {
              handleDelete(index);
            }
          "
        >-</span>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'UndoList',
  props: ['list'],
  data () {
    return {}
  },
  methods: {
    handleDelete (index) {
      this.$emit('delete', index)
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus"></style>
