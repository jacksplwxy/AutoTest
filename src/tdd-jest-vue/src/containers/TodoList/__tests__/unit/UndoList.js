import { shallowMount } from '@vue/test-utils'
import UndoList from '../../components/UndoList.vue'
import { findTestWrapper } from '../../../../utils/testUtils'

it('UndoList 参数为[]，count 值为0，且列表无内容', () => {
  const wrapper = shallowMount(UndoList, {
    propsData: { list: [] }
  })
  const countELem = findTestWrapper(wrapper, 'count')
  const listItems = findTestWrapper(wrapper, 'item')
  expect(countELem.at(0).text()).toEqual('0')
  expect(listItems.length).toEqual(0)
})

it('UndoList 参数为[1,2,3]，count 值为3，且列表有内容,且存在删除按钮', () => {
  const wrapper = shallowMount(UndoList, {
    propsData: { list: [1, 2, 3] }
  })
  const countELem = findTestWrapper(wrapper, 'count')
  const listItems = findTestWrapper(wrapper, 'item')
  const deleteButtons = findTestWrapper(wrapper, 'delete-button')
  expect(countELem.at(0).text()).toEqual('3')
  expect(listItems.length).toEqual(3)
  expect(deleteButtons.length).toEqual(3)
})

it('UndoList 删除按钮点击时，向外触发删除事件', () => {
  const wrapper = shallowMount(UndoList, {
    propsData: { list: [1, 2, 3] }
  })
  const deleteButton = findTestWrapper(wrapper, 'delete-button').at(1)
  deleteButton.trigger('click')
  expect(wrapper.emitted().delete).toBeTruthy()
  expect(wrapper.emitted().delete[0][0]).toBe(1)
})
