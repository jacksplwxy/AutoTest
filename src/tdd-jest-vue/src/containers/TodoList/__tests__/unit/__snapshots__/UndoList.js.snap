// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header样式发生改变做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text " data-test="input" placeholder="请输入待完成项" class="header-input">
  </div>
</div>
`;

exports[`UndoList 参数为[]，count 值为0，且列表无内容 1`] = `
<div>
  UndoList
</div>
`;
